map/src/menu/settings/DeleteFileVersionDialog.jsx

```diff
@@ -0,0 +1,52 @@
+import { Dialog } from '@material-ui/core';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogActions from '@mui/material/DialogActions';
+import { Button } from '@mui/material';
+import React, { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { deleteFile, deleteFileAllVersions, deleteFileVersion, formatString } from '../../manager/SettingsManager';
+import AppContext from '../../context/AppContext';
+
+export default function DeleteFileVersionDialog({
+    dialogOpen,
+    setDialogOpen,
+    file,
+    changes,
+    setChanges,
+    deleteVersion = false,
+    deleteAllVersions = false,
+}) {
+    const ctx = useContext(AppContext);
+    const { t } = useTranslation();
+    const toggleShowDialog = () => {
+        setDialogOpen(!dialogOpen);
+    };
+
+    async function deleteF() {
```

Comment:
When a function is actually "async", it is prettier to use `await child()` instead of child().then()